MaxText/input_pipeline/_hf_data_processing.py

```diff
@@ -0,0 +1,124 @@
+"""
+Copyright 2023 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+"""Input pipeline using Huggingface datasets."""
+
+import ml_collections
+import jax
+from datasets import load_dataset
+from transformers import AutoTokenizer
+import grain.python as grain
+
+from input_pipeline import _hf_operations
+from input_pipeline import _grain_operations
```

Comment:
Seems odd that hf depends on _grain_operations? Are these common operations that apply to both? If so could we move (or rename existing file) them to a file called something like "input_pipeline_utils" or "_common_operations".

This could be done in a followup PR

Reply:
They are now in _input_pipeline_utils.py